Units division now honors order of operation

cilvedson opened the pull request:
Before: `a/b/c = a*c/b`
After: `a/b/c = a/(b*c)`

The problem was that division was using recursion which means that division was starting at the right and moving to the left.

Primary: Dustin

cilvedson commented on src/test/java/javax/measure/unit/UnitTest.java:
```diff
@@ -1,11 +1,134 @@
+        map.put("N/kg*(m*h)", "N·m·h/kg");
+        map.put("N*kg/(m*h)", "N·kg/(m·h)");
+        map.put("N*kg*(m/h)", "N·kg·m/h");
+
```
The rest of the tests here didn't pass with my previous workaround.  (It wasn't a big concern because units of this structure are unlikely.)  They pass with this fix.

cilvedson commented on pom.xml:
```diff
@@ -5,7 +5,7 @@
     <artifactId>jscience</artifactId>
     <packaging>jar</packaging>
-    <version>4.3.2</version>
+    <version>4.3.4</version>
```
This will be 4.3.2 in the end.  (Before forking, the library was 4.3.1.)

cilvedson commented on src/main/java/javax/measure/unit/UnitFormat.java:
```diff
@@ -441,7 +443,9 @@ public String nameFor(Unit<?> unit) {
                             result = result.divide(d);
                         }
                     } else {
```
The problem and fix is here.

jrust commented:
Should this be targeted at the main repo or is plan to create our own version of the package?

cilvedson commented:
We'll maintain our own version.  The public project is dead.

srosenthal commented on src/main/java/javax/measure/unit/UnitFormat.java:
```diff
@@ -16,6 +16,7 @@
 import java.text.ParsePosition;
 import java.util.HashMap;
 import java.util.Locale;
+import java.util.Stack;
```
Super minor - looks unused?

srosenthal replied:
Nevermind, I'm bad at reading Github's diffs

srosenthal commented on src/main/java/javax/measure/unit/UnitFormat.java:
```diff
@@ -530,8 +534,31 @@ private void check(boolean expr, String message, CharSequence csq,
-        private Exponent readExponent (CharSequence csq, ParsePosition pos) {
+
+        private void checkForBalancedParentheses(CharSequence csq) throws ParseException {
+            Stack<Character> stack = new Stack<Character>();
```
Possible tweak: looks like you could use an int to count open-parens instead of a Stack

cilvedson commented:
This PR interface is painful compared to bitbucket and I can't figure out how to comment on other comments. My activity is this: 

- I updated the balanced paren check with Stephen's suggestion.
- I added comments to the fix area.
- I included some exponents in my division tests.

cilvedson commented:
P.S. Thanks to Dustin for setting up the forked work environment. It ended up being non-trivial.
